### PR TITLE
Added subscriber for MultiDOFJointTrajectory messages 

### LINKED ISSUE #243: add MultiDOFTrajectory as reference input for the controller
in addition to mav_msgs/CommandTrajectoryPositionYaw
http://docs.ros.org/indigo/api/trajectory_msgs/html/msg/MultiDOFJointTrajectory.html

### PR DESCRIPTION
**Pull request by markusachtelik**
and extended waypoint_publisher_file to test it. 
resolves #243

needs https://github.com/ethz-asl/mav_comm/pull/26

### CODE REVIEW

File: rotors_gazebo/src/waypoint_publisher_file.cpp [hunk 1]
```diff
@@ -60,12 +62,19 @@ int main(int argc, char** argv) {
...
-  if (args.size() != 2) {
+  if (args.size() != 2 && args.size() != 3) {
     ROS_ERROR(
-        "Usage: waypoint_publisher <waypoint_file> (one per line, space separated: wait_time [s] x[m] y[m] z[m] yaw[deg])");
+        "Usage: waypoint_publisher <waypoint_file> [--multi_dof_joint_trajectory](one per line, space separated: wait_time [s] x[m] y[m] z[m] yaw[deg])");
     return -1;
   }
 
+  bool use_multi_dof_joint_trajectory = false;
+  if(args.size() == 3){
```
Comment (ffurrer) [inline rotors_gazebo/src/waypoint_publisher_file.cpp]: spaces around if clause

File: rotors_gazebo/src/waypoint_publisher_file.cpp [hunk 2]
```diff
@@ -60,12 +62,19 @@ int main(int argc, char** argv) {
...
+    if(args[2] == "--multi_dof_joint_trajectory"){
```
Comment (ffurrer) [inline rotors_gazebo/src/waypoint_publisher_file.cpp]: same

**Comment by markusachtelik** [comment]
test this please.

File: rotors_gazebo/src/waypoint_publisher_file.cpp [hunk 3]
```diff
@@ -60,12 +62,19 @@ int main(int argc, char** argv) {
   ros::V_string args;
   ros::removeROSArgs(argc, argv, args);
 
-  if (args.size() != 2) {
+  if (args.size() != 2 && args.size() != 3) {
     ROS_ERROR(
-        "Usage: waypoint_publisher <waypoint_file> (one per line, space separated: wait_time [s] x[m] y[m] z[m] yaw[deg])");
+        "Usage: waypoint_publisher <waypoint_file> [--multi_dof_joint_trajectory](one per line, space separated: wait_time [s] x[m] y[m] z[m] yaw[deg])");
```
Comment (ffurrer) [inline rotors_gazebo/src/waypoint_publisher_file.cpp]: this is slightly confusing now, I rewrite this sentence.

**Comment by ffurrer** [comment]
lgtm

**Pull request merged**